plan/physical_plan_builder.go

```diff
@@ -351,6 +351,10 @@ func enforceProperty(prop *requiredProperty, info *physicalPlanInfo) *physicalPl
 	return info
 }
 
+// removeLimit removes limit from prop. For example, When handling Sort,Limit -> Selection, we can't pass the Limit
+// across the selection, because selection decreases the size of data, but we can pass the Sort below the selection. In
+// this case, we set removeAll true. When handling Limit(1,1) -> LeftOuterJoin, we can pass the limit across join's left
+// path, because the left outer join increases the size of data, but we can't pass offset value. So we set remove All to false.
```

Comment:
So when `removeAll` is set to true, it's meant to remove offset, then how about define a different function named `removeOffset`?